### DISCUSSION TITLE
Feature Request: More Variables in ProcessToRunOnBan

### DISCUSSION THREAD

**Discussion by Mr-Rayman** [DigitalRuby/IPBan#189] [Ideas]
Hi there,

thanks for your great software! 

In my ipban configuration I use 'ProcessToRunOnBan' with variable ###IPADDRESS### to call a script that sends an email whenever a block action occured.

It would be nice to know from what Source the blocking action came, since different sources have different criticality for me. So please consider providing an additional variable (e.g. ###SOURCE###) for 'ProcessToRunOnBan' that contains the source/log file in which the block event occoured. (Or maybe this exists already and I didn't see it in the documentation?)

Thanks!
Stefan

**Comment by jjxtra** [comment]
It should work as you describe. I'll update the docs.

**Reply by gotspatel** [reply]
sir is this updated in docs yet.

Regards

**Reply by gotspatel** [reply]
2023-01-20 06:38:38.7313|ERROR|IPBan|Error Exception: System.ArgumentException: Invalid config option for process to run: C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe|C:\IPBan\report_abuse_db.ps1|###IPADDRESS###|###SOURCE### -- should be two strings, | delimited with program and arguments.
   at DigitalRuby.IPBanCore.IPBanService.ExecuteExternalProcessForIPAddresses(String programToRun, IReadOnlyCollection`1 ipAddresses)
   at DigitalRuby.IPBanCore.IPBanService.ProcessPendingFailedLogins(IReadOnlyList`1 ipAddresses)

Can you please advise what may be wrong


C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe|C:\IPBan\report_abuse_db.ps1|###IPADDRESS###|###SOURCE###

**Reply by jjxtra** [reply]
Also I don't support the abuse ip db scripts and would recommend ipthreat as it is a 100% free service doing the same thing as abuseip db

**Comment by jjxtra** [comment]
Newline separate the processes I believe

**Reply by gotspatel** [reply]
I want to pass 2 arguments, is this the correct method?

**Reply by jjxtra** [reply]
Yes, separate each process|arguments pair with a newline.

**Reply by gotspatel** [reply]
Sir 

Is this correct? or need some change

`<add key="ProcessToRunOnBan" value="C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe|C:\IPBan\report_abuse_db.ps1|###IPADDRESS###|###SOURCE###"/>`

**Reply by jjxtra** [reply]
```xml
<add key="ProcessToRunOnBan" value="C:\Windows\system32\WindowsPowerShell\v1.0\powershell.exe|C:\IPBan\report_abuse_db.ps1 ###IPADDRESS### ###SOURCE###"/>
```

**Reply by jjxtra** [reply]
If that does not work, try

```xml
<add key="ProcessToRunOnBan" value="C:\IPBan\report_abuse_db.ps1|###IPADDRESS### ###SOURCE###"/>
```

**Reply by gotspatel** [reply]
sure thank you will update here which works

**Comment by gotspatel** [comment]
I am already contributing to ipthreat.net and this is not just for abuseipdb I am asking but a generic question that in case of passign multiple parameters is this the correct format/way

Regards
